### PR TITLE
Add genDocs function for mesheryctl

### LINKED ISSUE #5201: [Docs] Create autogeneration of docs for `mesheryctl` reference
#### Current State
`mesheryctl` reference is currently maintained manually by creating markdown files and YAML for data associated with it. So it needs to be periodically updated to avoid getting stale.

#### Desired State
Cobra CLI has inbuilt function to autogenerate doc pages with data present in each command file written in go. We need customized version in order to satisfy the layout we're following. Refer PR #5108 for work done so far. 

#### Notes
Generating markdown is easy but YAML is bit hard and requires extensive work to achieve the format like below files:

https://github.com/meshery/meshery/blob/417cd4b1316ec10448c2938ea46ca28194e8376d/docs/_data/mesheryctlcommands/cmds.yml#L1-L26

---

#### [Contributor Guides](https://docs.meshery.io/project/contributing) and Resources
- 📚 [Instructions for contributing to documentation](https://docs.meshery.io/project/contributing/contributing-docs)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/build-and-release)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by alphaX86**
Signed-off-by: Aadhitya A <aadhitya864@gmail.com>

**Description**

This PR fixes #5201

This PR is a work continued from #5108 

**Notes for Reviewers**
WIP

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.
 
Issues:
- [x] Routing needs to be fixed
- [x] All pages need to be stable
- [ ] Design (awaiting review)

Example tracker:
- [x] main
- [x] system
- [x] system-context
- [x] system-channel
- [x] perf
- [x] mesh
- [x] pattern
- [ ] filter

### CODE REVIEW

File: mesheryctl/internal/cli/root/root.go
Comment (alphaX86) [reply mesheryctl/internal/cli/root/root.go:52]: I'll show a demo webpage for this...

**Comment by alphaX86** [comment]
Note: The pages needs some work on after review, but as far as concerned except `mesheryctl` page (for now) others are linked with index.

We may also need YAML for index page else it'd get easily lost :)

**Comment by l5io** [comment]
🚀 Preview for commit ca284b5b7c002bdc725b434842c306f0b8657455 at: https://625af7498c9f153280c0584e--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 3732ea8508d8c363862fdc6eb999959b3b208040 at: https://625afbc1e76dcd35b5c2e9d1--meshery-docs-preview.netlify.app

**Comment by alphaX86** [comment]
> 🚀 Preview for commit [3732ea8](https://github.com/meshery/meshery/commit/3732ea8508d8c363862fdc6eb999959b3b208040) at: [625afbc1e76dcd35b5c2e9d1--meshery-docs-preview.netlify.app](https://625afbc1e76dcd35b5c2e9d1--meshery-docs-preview.netlify.app)

Seems the new created files are not overriding the old ones unlike in local variant...

**Comment by alphaX86** [comment]
image

image

**Comment by alphaX86** [comment]
image

image

**Comment by alphaX86** [comment]
I minimized the zoom for better view. The site preview isn't generating the files newly added

**Comment by l5io** [comment]
🚀 Preview for commit ca0dbf4c675ac6cb16ba3fa8c65e916b3c8735ad at: https://625bd26b37f5c33e788823e1--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 92a4645e0dbd0b3c967f1dbcfc32cbae09e74b96 at: https://625bf39094cca84ffe0629b2--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 69a0b700356be24ccc373d29e70e4d3bf54318be at: https://625bf5e5bef9c249919e6724--meshery-docs-preview.netlify.app

**Comment by alphaX86** [comment]
Even I removed old site files, it's being redirected to original one :(

**Comment by piyushsingariya** [comment]
> Even I removed old site files, it's being redirected to original one :(

Yup, not able to check how this looks. PLs fix 😭

**Comment by alphaX86** [comment]
😭😭😭

**Comment by alphaX86** [comment]
> > Even I removed old site files, it's being redirected to original one :(
> 
> Yup, not able to check how this looks. PLs fix 😭

Could you try looking on local version then? At least for now? :')

**Comment by l5io** [comment]
🚀 Preview for commit 53d63100477fc97d7b66516888a56e84007ac657 at: https://626167e2a81b4575f085d2b5--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 4ee37e41dd41284e8c2258caa4eb08913bba779f at: https://62618a4dbae9021df2aada98--meshery-docs-preview.netlify.app

**Comment by alphaX86** [comment]
It worked!! 😄

**Comment by l5io** [comment]
🚀 Preview for commit 02758873afe143fcddc5f91b9c649966949395ce at: https://62618da98fce8e1bcb9d27ca--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit a56a5fe53ff78264d2e53a510a6d5de78dc1b17a at: https://62618fd0c33d211e677773ed--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 84e33835980188244a83a36ab26d3f896a8ace77 at: https://626217bd9e21ec1392306283--meshery-docs-preview.netlify.app

**Comment by piyushsingariya** [comment]
@alphaX86 can you pull this off?
https://626217bd9e21ec1392306283--meshery-docs-preview.netlify.app/reference/mesheryctl/system/restart#see-also

Under See Also section.
We include a link to go back to parent command.
We include list of links to other sister commands of the command.
We include list of links to the sub-commands of the command.

**Comment by piyushsingariya** [comment]
Also I would like you to manually iterate each command's example written in code, add more and elaborated examples, make sure the quality of comments with the examples to be good and understandable.

With that I think we'll be almost at the verge of clearing off this goal from your internship deliverables. 

image

**Comment by alphaX86** [comment]
> @alphaX86 can you pull this off? https://626217bd9e21ec1392306283--meshery-docs-preview.netlify.app/reference/mesheryctl/system/restart#see-also
> 
> Under See Also section. We include a link to go back to parent command. We include list of links to other sister commands of the command. We include list of links to the sub-commands of the command.

That's where one issue started, link was able to generate successfully but its being appended so its like ref/mctl/system/context/main etc and so on... Other than that, I need to figure a way

**Comment by alphaX86** [comment]
> Also I would like you to manually iterate each command's example written in code, add more and elaborated examples, make sure the quality of comments with the examples to be good and understandable.
> 
> With that I think we'll be almost at the verge of clearing off this goal from your internship deliverables.
> 
> image

So they'd appear in CLI client app as well, is it OK for you to have extended examples in docs?

**Comment by piyushsingariya** [comment]
With adding examples in the command, I meant a list of commands have outdated examples and very few examples that doesn't cover the complete feature list of the command. So I asked you to make sure we're doing that.

**Comment by alphaX86** [comment]
Ok consider it drive

**Comment by leecalcote** [comment]
Nearly there... good work, @alphaX86

**Comment by l5io** [comment]
🚀 Preview for commit 8c2f394a6949c33385626f685b15d72bd1e9c1e9 at: https://626ac9754ba81305fec4fa27--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 7aad98f13a9b1cdd3db6828225a0c9d0b844700d at: https://626bf7157805700d3141c1e2--meshery-docs-preview.netlify.app

**Comment by leecalcote** [comment]
I didn’t see the GitHub workflow until now. Make sense.

**Comment by l5io** [comment]
🚀 Preview for commit d75aa32601614758dafa5e0bebcfb94687789135 at: https://6277ad220f4e573a012aa2c0--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 516688e1bb2934430d5af572ffa3f025a3f783b4 at: https://6277b1db9cbea0432f410177--meshery-docs-preview.netlify.app

**Comment by alphaX86** [comment]
Folks, the formatting is now fixed and stable :)

**Comment by l5io** [comment]
🚀 Preview for commit e287bd51b417b48fe985f3931d3e9aaad4473cc8 at: https://6277b359762cc143cc45043b--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 4a66f691bec56581adfc70d3a6d5533dea0c22ee at: https://62794fd94d8d51069f06119d--meshery-docs-preview.netlify.app

**Review by piyushsingariya — DISMISSED** [review]
LGTM

**Review by leecalcote — CHANGES_REQUESTED** [review]
Good to go here when `filter` examples are provided, and when autogeneration process is documented. When does this autogeneration occur? Please add to build and release docs section, identifying trigger.

One again, good job on mesheryctl docs, @alphaX86 
Onigger.

**Comment by alphaX86** [comment]
Auto generation CI will activate when PR is merged, that is when a user makes any changes to the CLI files.

For the filter subcommands, I'll add the examples today evening and change soon

**Comment by alphaX86** [comment]
We got a warning! ⚠️ `cmd/wrk2` seems to be missing and it messes with the docs build! 😨

**Comment by alphaX86** [comment]
@leecalcote we need to remove that submodule `cmd/wrk2` to work (since it's not clickable and empty) else it messes the entire docs build...

**Comment by alphaX86** [comment]
@warunicorn19 maybe the retrigger of the CI action uses cache of the data previously had :(

So I don't think we can check rerun now, unless if new changes made if I hope

**Comment by leecalcote** [comment]
Triggering new run of build checks....

**Comment by l5io** [comment]
🚀 Preview for commit 9bdd18702faa56d1874938073897a21d91ced748 at: https://627e8956891cbd06d638f1e4--meshery-docs-preview.netlify.app

**Comment by alphaX86** [comment]
Halt the PR merge... I'll fix the filter table quick

**Comment by l5io** [comment]
🚀 Preview for commit 7c63bafe764455ec5dec56c81eaf181416d2007f at: https://627e8f3085757e005d487a51--meshery-docs-preview.netlify.app

**Comment by alphaX86** [comment]
@warunicorn19 PR now available for final review. Let's tie this off and check CI action as well after merge ;)

**Comment by leecalcote** [comment]
Formatting issue with anchor. We need a slight change in URL reference for each subcommand
Screen Shot 2022-05-13 at 12 06 40 PM
.

**Comment by l5io** [comment]
🚀 Preview for commit 421c85d396d2e974cdcad8ec162522510d8b81f5 at: https://627fc828f8e00e7f943aeb4f--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit f891aac81ca1d816f1953d30af1fcedcdb570a1b at: https://6283b34e0fe4b81ebee5d1dc--meshery-docs-preview.netlify.app

**Comment by leecalcote** [comment]
Not just Safari, no, but happening in Chrome, too. Try this link: https://6283b34e0fe4b81ebee5d1dc--meshery-docs-preview.netlify.app/reference/mesheryctl/version

**Comment by alphaX86** [comment]
Brave:
image

Firefox:
image

No issues detected, @warunicorn19 any idea you about this?

**Pull request merged**
